llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -199,6 +207,15 @@ getVectorLoweringShape(EVT VectorEVT) {
   switch (VectorVT.SimpleTy) {
   default:
     return std::nullopt;
+  case MVT::v4i64:
```

Comment:
We could support `v2i128` as well right? I suppose this would require more invasive changes changes as this isn't an MVT, so I'd be fine leaving that to the side for now if you prefer.

Reply:
Adding b128 support may need more plumbing (we have a fair number of special cases for handling i128), which makes it a good candidate for a separate patch.

Reply:
Yeah, I think you're right, but I would prefer if we did this in a separate future change.